src/simplejavatexteditor/UI.java

```diff
@@ -630,13 +643,25 @@ public void drop(DropTargetDropEvent e) {
 
                         try {
                             String fileName = tr.getTransferData(flavors[i]).toString().replace("[", "").replace("]", "");
-                            FileInputStream fis = new FileInputStream(new File(fileName));
-                            byte[] ba = new byte[fis.available()];
-                            fis.read(ba);
-                            textArea.setText(new String(ba));
-                            fis.close();
-                        }
-                        catch (Exception ex) {
+                            String[] extensionFilter = {".txt", ".dat", ".log", ".xml", ".mf", ".html"}; // allowed file filter extentions for drag and drop
```

Comment:
Can this be a class attribute instead? We might have to reuse it later on elsewhere in the class... (and easier to find it if we have to add more file extensions). What do you think?

Reply:
Hi. yes of course. Is only an example. You can create a setup.class as well.

Have a nice day

Reply:
Yes, that was a suggestion 😉